app.js

```diff
@@ -17,15 +18,24 @@ app.get("/api/articles", getArticles);
 
 app.get("/api/articles/:article_id/comments", getComments);
 
+app.post("/api/articles/:article_id/comments", postComment);
+
+
+
 app.use((err, req, res, next) => {
   if (err.status && err.message) {
-    res.status(err.status).send({ message: err.message });
-  } else if (err.code === "22P02") {
-    res.status(400).send({ message: "Bad Request" });
-  } else {
-   console.log(err);
-    res.status(500).send({ message: "Server Error" });
+  res.status(err.status).send({ message: err.message });
+  }else if (err.code === "22P02") {
+  res.status(400).send({ message: "Bad Request" });
+  } else if (err.code === "23503" && err.constraint === "comments_article_id_fkey") {
+  res.status(404).send({ message: "article_id Does Not Exist" });
+  }
+  else if (err.code === "23502") {
+  res.status(400).send({ message: "Missing Required Fields" });
+  }else {
+  console.log(err);
+  res.status(500).send({ message: "Server Error" });
```

Comment:
All this is getting to be quite a lot of logic to be held by a single error handler, maybe at this point you could consider splitting them down into smaller error handlers, one for PSQL codes and another for your own custom errors ?